# Robert Corbett

Reddit post exploration part 2

This code is used to just find the number of posts I extracted from each subreddit. I have a large amount of data to choose from so I am only going to use a couple months of posts. The number of posts each subreddit has varies drastically. Some subreddits have hundreds of thousands of posts while some of a couple hundred. For the project, I want to use subreddits that have atleast 50,000 posts to work with.

In [1]:
import json
import numpy as np
from matplotlib import pyplot as plt
import nltk
import pandas as pd

I started by counting the number of lines in the json file for September, 2017.  I saved the total in a variable with the same name and printed the results.

In [2]:
path = "../../Reddit_Data_Trimmed/September_17/"

print("The number of posts in each subreddit for the month of September, 2017")

x = 0
with open(path + "Android.txt", 'r') as file_in:
    for line in file_in:
        x = x + 1
print("Len of Android: " + str(x))
Android = x

x = 0
with open(path + "Archaeology.txt", 'r') as file_in:
    for line in file_in:
        x = x + 1
print("Len of Archaeology: " + str(x))
Archaeology = x

x = 0
with open(path + "AskTrumpSupporters.txt", 'r') as file_in:
    for line in file_in:
        x = x + 1
print("Len of AskTrumpSupporters: " + str(x))
AskTrumpSupporters = x

x = 0
with open(path + "boardgames.txt", 'r') as file_in:
    for line in file_in:
        x = x + 1
print("Len of boardgames: " + str(x))
boardgames = x

x = 0
with open(path + "Conservative.txt", 'r') as file_in:
    for line in file_in:
        x = x + 1
print("Len of Conservative: " + str(x))
Conservative = x

x = 0
with open(path + "DCcomics.txt", 'r') as file_in:
    for line in file_in:
        x = x + 1
print("Len of DCcomics: " + str(x))
DCcomics = x

x = 0
with open(path + "food.txt", 'r') as file_in:
    for line in file_in:
        x = x + 1
print("Len of food: " + str(x))
food = x

x = 0
with open(path + "Futurism.txt", 'r') as file_in:
    for line in file_in:
        x = x + 1
print("Len of Futurism: " + str(x))
Futurism = x

x = 0
with open(path + "geopolitics.txt", 'r') as file_in:
    for line in file_in:
        x = x + 1
print("Len of geopolitics: " + str(x))
geopolitics = x

x = 0
with open(path + "gunpolitics.txt", 'r') as file_in:
    for line in file_in:
        x = x + 1
print("Len of gunpolitics: " + str(x))
gunpolitics = x

x = 0
with open(path + "headphones.txt", 'r') as file_in:
    for line in file_in:
        x = x + 1
print("Len of headphones: " + str(x))
headphones = x

x = 0
with open(path + "hockey.txt", 'r') as file_in:
    for line in file_in:
        x = x + 1
print("Len of hockey: " + str(x))
hockey = x

x = 0
with open(path + "indie_rock.txt", 'r') as file_in:
    for line in file_in:
        x = x + 1
print("Len of indie_rock: " + str(x))
indie_rock = x

x = 0
with open(path + "indieheads.txt", 'r') as file_in:
    for line in file_in:
        x = x + 1
print("Len of indieheads: " + str(x))
indieheads = x

x = 0
with open(path + "Libertarian.txt", 'r') as file_in:
    for line in file_in:
        x = x + 1
print("Len of Libertarian: " + str(x))
Libertarian = x

x = 0
with open(path + "MarchAgainstTrump.txt", 'r') as file_in:
    for line in file_in:
        x = x + 1
print("Len of MarchAgainstTrump: " + str(x))
MarchAgainstTrump = x

x = 0
with open(path + "neoliberal.txt", 'r') as file_in:
    for line in file_in:
        x = x + 1
print("Len of neoliberal: " + str(x))
neoliberal = x

x = 0
with open(path + "NeverTrump.txt", 'r') as file_in:
    for line in file_in:
        x = x + 1
print("Len of NeverTrump: " + str(x))
NeverTrump = x

x = 0
with open(path + "photography.txt", 'r') as file_in:
    for line in file_in:
        x = x + 1
print("Len of photography: " + str(x))
photography = x

x = 0
with open(path + "politics.txt", 'r') as file_in:
    for line in file_in:
        x = x + 1
print("Len of politics: " + str(x))
politics = x

x = 0
with open(path + "PoliticsWithoutTheBan.txt", 'r') as file_in:
    for line in file_in:
        x = x + 1
print("Len of PoliticsWithoutTheBan: " + str(x))
PoliticsWithoutTheBan = x

x = 0
with open(path + "seinfeld.txt", 'r') as file_in:
    for line in file_in:
        x = x + 1
print("Len of seinfeld: " + str(x))
seinfeld = x

x = 0
with open(path + "The_Donald.txt", 'r') as file_in:
    for line in file_in:
        x = x + 1
print("Len of The_Donald: " + str(x))
The_Donald = x

x = 0
with open(path + "worldnews.txt", 'r') as file_in:
    for line in file_in:
        x = x + 1
print("Len of worldnews: " + str(x))
worldnews = x

The number of posts in each subreddit for the month of September, 2017
Len of Android: 165509
Len of Archaeology: 574
Len of AskTrumpSupporters: 41290
Len of boardgames: 61240
Len of Conservative: 55766
Len of DCcomics: 24647
Len of food: 69709
Len of Futurism: 197
Len of geopolitics: 7313
Len of gunpolitics: 4390
Len of headphones: 29723
Len of hockey: 199269
Len of indie_rock: 491
Len of indieheads: 35572
Len of Libertarian: 103633
Len of MarchAgainstTrump: 11761
Len of neoliberal: 212190
Len of NeverTrump: 180
Len of photography: 36431
Len of politics: 1390463
Len of PoliticsWithoutTheBan: 0
Len of seinfeld: 5277
Len of The_Donald: 904159
Len of worldnews: 777276


I did the same for August, 2017 but added the totals to the totals for September and printed the totals for the month below.

In [3]:
path = "../../Reddit_Data_Trimmed/August_17/"

print("The number of posts in each subreddit for the month of August, 2017")

x = 0
with open(path + "Android.txt", 'r') as file_in:
    for line in file_in:
        x = x + 1
print("Len of Android: " + str(x))
Android = Android + x

x = 0
with open(path + "Archaeology.txt", 'r') as file_in:
    for line in file_in:
        x = x + 1
print("Len of Archaeology: " + str(x))
Archaeology = Archaeology + x

x = 0
with open(path + "AskTrumpSupporters.txt", 'r') as file_in:
    for line in file_in:
        x = x + 1
print("Len of AskTrumpSupporters: " + str(x))
AskTrumpSupporters = AskTrumpSupporters + x

x = 0
with open(path + "boardgames.txt", 'r') as file_in:
    for line in file_in:
        x = x + 1
print("Len of boardgames: " + str(x))
boardgames = boardgames + x

x = 0
with open(path + "Conservative.txt", 'r') as file_in:
    for line in file_in:
        x = x + 1
print("Len of Conservative: " + str(x))
Conservative = Conservative + x

x = 0
with open(path + "DCcomics.txt", 'r') as file_in:
    for line in file_in:
        x = x + 1
print("Len of DCcomics: " + str(x))
DCcomics = DCcomics + x

x = 0
with open(path + "food.txt", 'r') as file_in:
    for line in file_in:
        x = x + 1
print("Len of food: " + str(x))
food = food + x

x = 0
with open(path + "Futurism.txt", 'r') as file_in:
    for line in file_in:
        x = x + 1
print("Len of Futurism: " + str(x))
Futurism = Futurism + x

x = 0
with open(path + "geopolitics.txt", 'r') as file_in:
    for line in file_in:
        x = x + 1
print("Len of geopolitics: " + str(x))
geopolitics = geopolitics + x

x = 0
with open(path + "gunpolitics.txt", 'r') as file_in:
    for line in file_in:
        x = x + 1
print("Len of gunpolitics: " + str(x))
gunpolitics = gunpolitics + x

x = 0
with open(path + "headphones.txt", 'r') as file_in:
    for line in file_in:
        x = x + 1
print("Len of headphones: " + str(x))
headphones = headphones + x

x = 0
with open(path + "hockey.txt", 'r') as file_in:
    for line in file_in:
        x = x + 1
print("Len of hockey: " + str(x))
hockey = hockey + x

x = 0
with open(path + "indie_rock.txt", 'r') as file_in:
    for line in file_in:
        x = x + 1
print("Len of indie_rock: " + str(x))
indie_rock = indie_rock + x

x = 0
with open(path + "indieheads.txt", 'r') as file_in:
    for line in file_in:
        x = x + 1
print("Len of indieheads: " + str(x))
indieheads = indieheads + x

x = 0
with open(path + "Libertarian.txt", 'r') as file_in:
    for line in file_in:
        x = x + 1
print("Len of Libertarian: " + str(x))
Libertarian = Libertarian + x

x = 0
with open(path + "MarchAgainstTrump.txt", 'r') as file_in:
    for line in file_in:
        x = x + 1
print("Len of MarchAgainstTrump: " + str(x))
MarchAgainstTrump = MarchAgainstTrump + x

x = 0
with open(path + "neoliberal.txt", 'r') as file_in:
    for line in file_in:
        x = x + 1
print("Len of neoliberal: " + str(x))
neoliberal = neoliberal + x

x = 0
with open(path + "NeverTrump.txt", 'r') as file_in:
    for line in file_in:
        x = x + 1
print("Len of NeverTrump: " + str(x))
NeverTrump = NeverTrump + x

x = 0
with open(path + "photography.txt", 'r') as file_in:
    for line in file_in:
        x = x + 1
print("Len of photography: " + str(x))
photography = photography + x

x = 0
with open(path + "politics.txt", 'r') as file_in:
    for line in file_in:
        x = x + 1
print("Len of politics: " + str(x))
politics = politics + x

x = 0
with open(path + "PoliticsWithoutTheBan.txt", 'r') as file_in:
    for line in file_in:
        x = x + 1
print("Len of PoliticsWithoutTheBan: " + str(x))
PoliticsWithoutTheBan = PoliticsWithoutTheBan + x

x = 0
with open(path + "seinfeld.txt", 'r') as file_in:
    for line in file_in:
        x = x + 1
print("Len of seinfeld: " + str(x))
seinfeld = seinfeld + x

x = 0
with open(path + "The_Donald.txt", 'r') as file_in:
    for line in file_in:
        x = x + 1
print("Len of The_Donald: " + str(x))
The_Donald = The_Donald + x

x = 0
with open(path + "worldnews.txt", 'r') as file_in:
    for line in file_in:
        x = x + 1
print("Len of worldnews: " + str(x))
worldnews = worldnews + x

The number of posts in each subreddit for the month of August, 2017
Len of Android: 143884
Len of Archaeology: 805
Len of AskTrumpSupporters: 45575
Len of boardgames: 60254
Len of Conservative: 66364
Len of DCcomics: 27988
Len of food: 68758
Len of Futurism: 187
Len of geopolitics: 6269
Len of gunpolitics: 2836
Len of headphones: 30372
Len of hockey: 157833
Len of indie_rock: 320
Len of indieheads: 39867
Len of Libertarian: 94092
Len of MarchAgainstTrump: 19208
Len of neoliberal: 184248
Len of NeverTrump: 322
Len of photography: 41420
Len of politics: 1824215
Len of PoliticsWithoutTheBan: 10
Len of seinfeld: 4045
Len of The_Donald: 1057483
Len of worldnews: 741492


The data for July was corrupted and I could not get the json entries to open.  So I skipped July and am going to use June instead.  I do not see skipping July causing any problems with the project.

In [4]:
path = "../../Reddit_Data_Trimmed/June_17/"

print("The number of posts in each subreddit for the month of June, 2017")
x = 0
with open(path + "Android.txt", 'r') as file_in:
    for line in file_in:
        x = x + 1
print("Len of Android: " + str(x))
Android = Android + x

x = 0
with open(path + "Archaeology.txt", 'r') as file_in:
    for line in file_in:
        x = x + 1
print("Len of Archaeology: " + str(x))
Archaeology = Archaeology + x

x = 0
with open(path + "AskTrumpSupporters.txt", 'r') as file_in:
    for line in file_in:
        x = x + 1
print("Len of AskTrumpSupporters: " + str(x))
AskTrumpSupporters = AskTrumpSupporters + x

x = 0
with open(path + "boardgames.txt", 'r') as file_in:
    for line in file_in:
        x = x + 1
print("Len of boardgames: " + str(x))
boardgames = boardgames + x

x = 0
with open(path + "Conservative.txt", 'r') as file_in:
    for line in file_in:
        x = x + 1
print("Len of Conservative: " + str(x))
Conservative = Conservative + x

x = 0
with open(path + "DCcomics.txt", 'r') as file_in:
    for line in file_in:
        x = x + 1
print("Len of DCcomics: " + str(x))
DCcomics = DCcomics + x

x = 0
with open(path + "food.txt", 'r') as file_in:
    for line in file_in:
        x = x + 1
print("Len of food: " + str(x))
food = food + x

x = 0
with open(path + "Futurism.txt", 'r') as file_in:
    for line in file_in:
        x = x + 1
print("Len of Futurism: " + str(x))
Futurism = Futurism + x

x = 0
with open(path + "geopolitics.txt", 'r') as file_in:
    for line in file_in:
        x = x + 1
print("Len of geopolitics: " + str(x))
geopolitics = geopolitics + x

x = 0
with open(path + "gunpolitics.txt", 'r') as file_in:
    for line in file_in:
        x = x + 1
print("Len of gunpolitics: " + str(x))
gunpolitics = gunpolitics + x

x = 0
with open(path + "headphones.txt", 'r') as file_in:
    for line in file_in:
        x = x + 1
print("Len of headphones: " + str(x))
headphones = headphones + x

x = 0
with open(path + "hockey.txt", 'r') as file_in:
    for line in file_in:
        x = x + 1
print("Len of hockey: " + str(x))
hockey = hockey + x

x = 0
with open(path + "indie_rock.txt", 'r') as file_in:
    for line in file_in:
        x = x + 1
print("Len of indie_rock: " + str(x))
indie_rock = indie_rock + x

x = 0
with open(path + "indieheads.txt", 'r') as file_in:
    for line in file_in:
        x = x + 1
print("Len of indieheads: " + str(x))
indieheads = indieheads + x

x = 0
with open(path + "Libertarian.txt", 'r') as file_in:
    for line in file_in:
        x = x + 1
print("Len of Libertarian: " + str(x))
Libertarian = Libertarian + x

x = 0
with open(path + "MarchAgainstTrump.txt", 'r') as file_in:
    for line in file_in:
        x = x + 1
print("Len of MarchAgainstTrump: " + str(x))
MarchAgainstTrump = MarchAgainstTrump + x

x = 0
with open(path + "neoliberal.txt", 'r') as file_in:
    for line in file_in:
        x = x + 1
print("Len of neoliberal: " + str(x))
neoliberal = neoliberal + x

x = 0
with open(path + "NeverTrump.txt", 'r') as file_in:
    for line in file_in:
        x = x + 1
print("Len of NeverTrump: " + str(x))
NeverTrump = NeverTrump + x

x = 0
with open(path + "photography.txt", 'r') as file_in:
    for line in file_in:
        x = x + 1
print("Len of photography: " + str(x))
photography = photography + x

x = 0
with open(path + "politics.txt", 'r') as file_in:
    for line in file_in:
        x = x + 1
print("Len of politics: " + str(x))
politics = politics + x

x = 0
with open(path + "PoliticsWithoutTheBan.txt", 'r') as file_in:
    for line in file_in:
        x = x + 1
print("Len of PoliticsWithoutTheBan: " + str(x))
PoliticsWithoutTheBan = PoliticsWithoutTheBan + x

x = 0
with open(path + "seinfeld.txt", 'r') as file_in:
    for line in file_in:
        x = x + 1
print("Len of seinfeld: " + str(x))
seinfeld = seinfeld + x

x = 0
with open(path + "The_Donald.txt", 'r') as file_in:
    for line in file_in:
        x = x + 1
print("Len of The_Donald: " + str(x))
The_Donald = The_Donald + x

x = 0
with open(path + "worldnews.txt", 'r') as file_in:
    for line in file_in:
        x = x + 1
print("Len of worldnews: " + str(x))
worldnews = worldnews + x

The number of posts in each subreddit for the month of June, 2017
Len of Android: 124963
Len of Archaeology: 524
Len of AskTrumpSupporters: 44644
Len of boardgames: 63268
Len of Conservative: 47492
Len of DCcomics: 33775
Len of food: 67223
Len of Futurism: 161
Len of geopolitics: 6086
Len of gunpolitics: 3777
Len of headphones: 26346
Len of hockey: 486157
Len of indie_rock: 250
Len of indieheads: 39393
Len of Libertarian: 83183
Len of MarchAgainstTrump: 67415
Len of neoliberal: 160842
Len of NeverTrump: 327
Len of photography: 40423
Len of politics: 1818730
Len of PoliticsWithoutTheBan: 0
Len of seinfeld: 3882
Len of The_Donald: 1158567
Len of worldnews: 909968


Finally, below I printed the totals for each subreddit for the three months.

In [5]:
print("The number of posts in each subreddit for both months")
print("Android: " + str(Android))
print("Archaeology: " + str(Archaeology))
print("AskTrumpSupporters: " + str(AskTrumpSupporters))
print("boardgames: " + str(boardgames))
print("Conservative: " + str(Conservative))
print("DCcomics: " + str(DCcomics))
print("food: " + str(food))
print("Futurism: " + str(Futurism))
print("geopolitics: " + str(geopolitics))
print("gunpolitics: " + str(gunpolitics))
print("headphones: " + str(headphones))
print("hockey: " + str(hockey))
print("indie_rock: " + str(indie_rock))
print("indieheads: " + str(indieheads))
print("Libertarian: " + str(Libertarian))
print("MarchAgainstTrump: " + str(MarchAgainstTrump))
print("neoliberal: " + str(neoliberal))
print("NeverTrump: " + str(NeverTrump))
print("photography: " + str(photography))
print("politics: " + str(politics))
print("PoliticsWithoutTheBan: " + str(PoliticsWithoutTheBan))
print("seinfeld: " + str(seinfeld))
print("The_Donald: " + str(The_Donald))
print("worldnews: " + str(worldnews))

The number of posts in each subreddit for both months
Android: 434356
Archaeology: 1903
AskTrumpSupporters: 131509
boardgames: 184762
Conservative: 169622
DCcomics: 86410
food: 205690
Futurism: 545
geopolitics: 19668
gunpolitics: 11003
headphones: 86441
hockey: 843259
indie_rock: 1061
indieheads: 114832
Libertarian: 280908
MarchAgainstTrump: 98384
neoliberal: 557280
NeverTrump: 829
photography: 118274
politics: 5033408
PoliticsWithoutTheBan: 10
seinfeld: 13204
The_Donald: 3120209
worldnews: 2428736


For the project, I want political subreddits, both with and without political biases.  So, for biased political subreddits, I am going to use "Conservative", "Libertarian" and "neoliberal".  For nonbiased political subreddits, I am going to use "politics" and "worldnews".  As a control, I will use non political subreddits.  I will use "Android", "boardgames" and "hockey".

Next, I want to find out the average length of posts.  Many of the posts are very short.  I want to use the longest posts I can while mantaining the 50,000 post goal.  Below, I calculated the average lengths and how many posts are greater than 50, 100, 150, 200, 250, and 300 characters

In [6]:
num_posts = 0
num_chars = 0
over_50 = 0
over_100 = 0
over_150 = 0
over_200 = 0
over_250 = 0
over_300 = 0

with open("../../Reddit_Data_trimmed/September_17/Conservative.txt", 'r') as file_in:
    for line in file_in:
        line_in = json.loads(line)
        num_chars = num_chars + len(line_in['body'])
        if len(line_in['body'])>=50:
            over_50 = over_50 + 1
        if len(line_in['body'])>=100:
            over_100 = over_100 + 1
        if len(line_in['body'])>=150:
            over_150 = over_150 + 1
        if len(line_in['body'])>=200:
            over_200 = over_200 + 1
        if len(line_in['body'])>=250:
            over_250 = over_250 + 1
        if len(line_in['body'])>=300:
            over_300 = over_300 + 1            
        
        num_posts = num_posts + 1

with open("../../Reddit_Data_trimmed/August_17/Conservative.txt", 'r') as file_in:
    for line in file_in:
        line_in = json.loads(line)
        num_chars = num_chars + len(line_in['body'])
        if len(line_in['body'])>=50:
            over_50 = over_50 + 1
        if len(line_in['body'])>=100:
            over_100 = over_100 + 1
        if len(line_in['body'])>=150:
            over_150 = over_150 + 1
        if len(line_in['body'])>=200:
            over_200 = over_200 + 1
        if len(line_in['body'])>=250:
            over_250 = over_250 + 1
        if len(line_in['body'])>=300:
            over_300 = over_300 + 1   
                           
        num_posts = num_posts + 1
            
with open("../../Reddit_Data_trimmed/June_17/Conservative.txt", 'r') as file_in:
    for line in file_in:
        line_in = json.loads(line)
        num_chars = num_chars + len(line_in['body'])
        if len(line_in['body'])>=50:
            over_50 = over_50 + 1
        if len(line_in['body'])>=100:
            over_100 = over_100 + 1
        if len(line_in['body'])>=150:
            over_150 = over_150 + 1
        if len(line_in['body'])>=200:
            over_200 = over_200 + 1
        if len(line_in['body'])>=250:
            over_250 = over_250 + 1
        if len(line_in['body'])>=300:
            over_300 = over_300 + 1            

        num_posts = num_posts + 1

print("Number of posts in Conservative     : " + str(num_posts))
print("Average length of body              : " + str(num_chars/num_posts))
print("number of posts over 50   characters: " + str(over_50))
print("number of posts over 100  characters: " + str(over_100))
print("number of posts over 150  characters: " + str(over_150))
print("number of posts over 200  characters: " + str(over_200))
print("number of posts over 250  characters: " + str(over_250))
print("number of posts over 300  characters: " + str(over_300))

Number of posts in Conservative     : 169622
Average length of body              : 218.35662826755964
number of posts over 50   characters: 122319
number of posts over 100  characters: 91978
number of posts over 150  characters: 69784
number of posts over 200  characters: 54493
number of posts over 250  characters: 43749
number of posts over 300  characters: 35778


In [7]:
num_posts = 0
num_chars = 0
over_50 = 0
over_100 = 0
over_150 = 0
over_200 = 0
over_250 = 0
over_300 = 0

with open("../../Reddit_Data_trimmed/September_17/Libertarian.txt", 'r') as file_in:
    for line in file_in:
        line_in = json.loads(line)
        num_chars = num_chars + len(line_in['body'])
        if len(line_in['body'])>=50:
            over_50 = over_50 + 1
        if len(line_in['body'])>=100:
            over_100 = over_100 + 1
        if len(line_in['body'])>=150:
            over_150 = over_150 + 1
        if len(line_in['body'])>=200:
            over_200 = over_200 + 1
        if len(line_in['body'])>=250:
            over_250 = over_250 + 1
        if len(line_in['body'])>=300:
            over_300 = over_300 + 1            
        
        num_posts = num_posts + 1

with open("../../Reddit_Data_trimmed/August_17/Libertarian.txt", 'r') as file_in:
    for line in file_in:
        line_in = json.loads(line)
        num_chars = num_chars + len(line_in['body'])
        if len(line_in['body'])>=50:
            over_50 = over_50 + 1
        if len(line_in['body'])>=100:
            over_100 = over_100 + 1
        if len(line_in['body'])>=150:
            over_150 = over_150 + 1
        if len(line_in['body'])>=200:
            over_200 = over_200 + 1
        if len(line_in['body'])>=250:
            over_250 = over_250 + 1
        if len(line_in['body'])>=300:
            over_300 = over_300 + 1   
                           
        num_posts = num_posts + 1
            
with open("../../Reddit_Data_trimmed/June_17/Libertarian.txt", 'r') as file_in:
    for line in file_in:
        line_in = json.loads(line)
        num_chars = num_chars + len(line_in['body'])
        if len(line_in['body'])>=50:
            over_50 = over_50 + 1
        if len(line_in['body'])>=100:
            over_100 = over_100 + 1
        if len(line_in['body'])>=150:
            over_150 = over_150 + 1
        if len(line_in['body'])>=200:
            over_200 = over_200 + 1
        if len(line_in['body'])>=250:
            over_250 = over_250 + 1
        if len(line_in['body'])>=300:
            over_300 = over_300 + 1            

        num_posts = num_posts + 1

print("Number of posts in Libertarian      : " + str(num_posts))
print("Average length of body              : " + str(num_chars/num_posts))
print("number of posts over 50   characters: " + str(over_50))
print("number of posts over 100  characters: " + str(over_100))
print("number of posts over 150  characters: " + str(over_150))
print("number of posts over 200  characters: " + str(over_200))
print("number of posts over 250  characters: " + str(over_250))
print("number of posts over 300  characters: " + str(over_300))

Number of posts in Libertarian      : 280908
Average length of body              : 287.30290344169623
number of posts over 50   characters: 227488
number of posts over 100  characters: 177428
number of posts over 150  characters: 139855
number of posts over 200  characters: 112474
number of posts over 250  characters: 92363
number of posts over 300  characters: 77299


In [8]:
num_posts = 0
num_chars = 0
over_50 = 0
over_100 = 0
over_150 = 0
over_200 = 0
over_250 = 0
over_300 = 0

with open("../../Reddit_Data_trimmed/September_17/neoliberal.txt", 'r') as file_in:
    for line in file_in:
        line_in = json.loads(line)
        num_chars = num_chars + len(line_in['body'])
        if len(line_in['body'])>=50:
            over_50 = over_50 + 1
        if len(line_in['body'])>=100:
            over_100 = over_100 + 1
        if len(line_in['body'])>=150:
            over_150 = over_150 + 1
        if len(line_in['body'])>=200:
            over_200 = over_200 + 1
        if len(line_in['body'])>=250:
            over_250 = over_250 + 1
        if len(line_in['body'])>=300:
            over_300 = over_300 + 1            
        
        num_posts = num_posts + 1

with open("../../Reddit_Data_trimmed/August_17/neoliberal.txt", 'r') as file_in:
    for line in file_in:
        line_in = json.loads(line)
        num_chars = num_chars + len(line_in['body'])
        if len(line_in['body'])>=50:
            over_50 = over_50 + 1
        if len(line_in['body'])>=100:
            over_100 = over_100 + 1
        if len(line_in['body'])>=150:
            over_150 = over_150 + 1
        if len(line_in['body'])>=200:
            over_200 = over_200 + 1
        if len(line_in['body'])>=250:
            over_250 = over_250 + 1
        if len(line_in['body'])>=300:
            over_300 = over_300 + 1   
                           
        num_posts = num_posts + 1
            
with open("../../Reddit_Data_trimmed/June_17/neoliberal.txt", 'r') as file_in:
    for line in file_in:
        line_in = json.loads(line)
        num_chars = num_chars + len(line_in['body'])
        if len(line_in['body'])>=50:
            over_50 = over_50 + 1
        if len(line_in['body'])>=100:
            over_100 = over_100 + 1
        if len(line_in['body'])>=150:
            over_150 = over_150 + 1
        if len(line_in['body'])>=200:
            over_200 = over_200 + 1
        if len(line_in['body'])>=250:
            over_250 = over_250 + 1
        if len(line_in['body'])>=300:
            over_300 = over_300 + 1            

        num_posts = num_posts + 1

print(num_posts)
print("Average length of body: " + str(num_chars/num_posts))
print("number of posts over 50  characters: " + str(over_50))
print("number of posts over 100  characters: " + str(over_100))
print("number of posts over 150 characters: " + str(over_150))
print("number of posts over 200  characters: " + str(over_200))
print("number of posts over 250  characters: " + str(over_250))
print("number of posts over 300 characters: " + str(over_300))

557280
Average length of body: 153.20315281366638
number of posts over 50  characters: 350206
number of posts over 100  characters: 221754
number of posts over 150 characters: 150061
number of posts over 200  characters: 108318
number of posts over 250  characters: 82134
number of posts over 300 characters: 64611


In [9]:
num_posts = 0
num_chars = 0
over_50 = 0
over_100 = 0
over_150 = 0
over_200 = 0
over_250 = 0
over_300 = 0

with open("../../Reddit_Data_trimmed/September_17/politics.txt", 'r') as file_in:
    for line in file_in:
        line_in = json.loads(line)
        num_chars = num_chars + len(line_in['body'])
        if len(line_in['body'])>=50:
            over_50 = over_50 + 1
        if len(line_in['body'])>=100:
            over_100 = over_100 + 1
        if len(line_in['body'])>=150:
            over_150 = over_150 + 1
        if len(line_in['body'])>=200:
            over_200 = over_200 + 1
        if len(line_in['body'])>=250:
            over_250 = over_250 + 1
        if len(line_in['body'])>=300:
            over_300 = over_300 + 1            
        
        num_posts = num_posts + 1

with open("../../Reddit_Data_trimmed/August_17/politics.txt", 'r') as file_in:
    for line in file_in:
        line_in = json.loads(line)
        num_chars = num_chars + len(line_in['body'])
        if len(line_in['body'])>=50:
            over_50 = over_50 + 1
        if len(line_in['body'])>=100:
            over_100 = over_100 + 1
        if len(line_in['body'])>=150:
            over_150 = over_150 + 1
        if len(line_in['body'])>=200:
            over_200 = over_200 + 1
        if len(line_in['body'])>=250:
            over_250 = over_250 + 1
        if len(line_in['body'])>=300:
            over_300 = over_300 + 1   
                           
        num_posts = num_posts + 1
            
with open("../../Reddit_Data_trimmed/June_17/politics.txt", 'r') as file_in:
    for line in file_in:
        line_in = json.loads(line)
        num_chars = num_chars + len(line_in['body'])
        if len(line_in['body'])>=50:
            over_50 = over_50 + 1
        if len(line_in['body'])>=100:
            over_100 = over_100 + 1
        if len(line_in['body'])>=150:
            over_150 = over_150 + 1
        if len(line_in['body'])>=200:
            over_200 = over_200 + 1
        if len(line_in['body'])>=250:
            over_250 = over_250 + 1
        if len(line_in['body'])>=300:
            over_300 = over_300 + 1            

        num_posts = num_posts + 1

print(num_posts)
print("Average length of body: " + str(num_chars/num_posts))
print("number of posts over 50  characters: " + str(over_50))
print("number of posts over 100  characters: " + str(over_100))
print("number of posts over 150 characters: " + str(over_150))
print("number of posts over 200  characters: " + str(over_200))
print("number of posts over 250  characters: " + str(over_250))
print("number of posts over 300 characters: " + str(over_300))

5033408
Average length of body: 202.61892399741885
number of posts over 50  characters: 3629278
number of posts over 100  characters: 2579653
number of posts over 150 characters: 1897668
number of posts over 200  characters: 1456237
number of posts over 250  characters: 1155323
number of posts over 300 characters: 941578


In [10]:
num_posts = 0
num_chars = 0
over_50 = 0
over_100 = 0
over_150 = 0
over_200 = 0
over_250 = 0
over_300 = 0

with open("../../Reddit_Data_trimmed/September_17/worldnews.txt", 'r') as file_in:
    for line in file_in:
        line_in = json.loads(line)
        num_chars = num_chars + len(line_in['body'])
        if len(line_in['body'])>=50:
            over_50 = over_50 + 1
        if len(line_in['body'])>=100:
            over_100 = over_100 + 1
        if len(line_in['body'])>=150:
            over_150 = over_150 + 1
        if len(line_in['body'])>=200:
            over_200 = over_200 + 1
        if len(line_in['body'])>=250:
            over_250 = over_250 + 1
        if len(line_in['body'])>=300:
            over_300 = over_300 + 1            
        
        num_posts = num_posts + 1

with open("../../Reddit_Data_trimmed/August_17/worldnews.txt", 'r') as file_in:
    for line in file_in:
        line_in = json.loads(line)
        num_chars = num_chars + len(line_in['body'])
        if len(line_in['body'])>=50:
            over_50 = over_50 + 1
        if len(line_in['body'])>=100:
            over_100 = over_100 + 1
        if len(line_in['body'])>=150:
            over_150 = over_150 + 1
        if len(line_in['body'])>=200:
            over_200 = over_200 + 1
        if len(line_in['body'])>=250:
            over_250 = over_250 + 1
        if len(line_in['body'])>=300:
            over_300 = over_300 + 1   
                           
        num_posts = num_posts + 1
            
with open("../../Reddit_Data_trimmed/June_17/worldnews.txt", 'r') as file_in:
    for line in file_in:
        line_in = json.loads(line)
        num_chars = num_chars + len(line_in['body'])
        if len(line_in['body'])>=50:
            over_50 = over_50 + 1
        if len(line_in['body'])>=100:
            over_100 = over_100 + 1
        if len(line_in['body'])>=150:
            over_150 = over_150 + 1
        if len(line_in['body'])>=200:
            over_200 = over_200 + 1
        if len(line_in['body'])>=250:
            over_250 = over_250 + 1
        if len(line_in['body'])>=300:
            over_300 = over_300 + 1            

        num_posts = num_posts + 1

print(num_posts)
print("Average length of body: " + str(num_chars/num_posts))
print("number of posts over 50  characters: " + str(over_50))
print("number of posts over 100  characters: " + str(over_100))
print("number of posts over 150 characters: " + str(over_150))
print("number of posts over 200  characters: " + str(over_200))
print("number of posts over 250  characters: " + str(over_250))
print("number of posts over 300 characters: " + str(over_300))

2428736
Average length of body: 200.97222794078894
number of posts over 50  characters: 1691007
number of posts over 100  characters: 1215445
number of posts over 150 characters: 901245
number of posts over 200  characters: 693553
number of posts over 250  characters: 548246
number of posts over 300 characters: 444789


In [11]:
num_posts = 0
num_chars = 0
over_50 = 0
over_100 = 0
over_150 = 0
over_200 = 0
over_250 = 0
over_300 = 0

with open("../../Reddit_Data_trimmed/September_17/Android.txt", 'r') as file_in:
    for line in file_in:
        line_in = json.loads(line)
        num_chars = num_chars + len(line_in['body'])
        if len(line_in['body'])>=50:
            over_50 = over_50 + 1
        if len(line_in['body'])>=100:
            over_100 = over_100 + 1
        if len(line_in['body'])>=150:
            over_150 = over_150 + 1
        if len(line_in['body'])>=200:
            over_200 = over_200 + 1
        if len(line_in['body'])>=250:
            over_250 = over_250 + 1
        if len(line_in['body'])>=300:
            over_300 = over_300 + 1            
        
        num_posts = num_posts + 1

with open("../../Reddit_Data_trimmed/August_17/Android.txt", 'r') as file_in:
    for line in file_in:
        line_in = json.loads(line)
        num_chars = num_chars + len(line_in['body'])
        if len(line_in['body'])>=50:
            over_50 = over_50 + 1
        if len(line_in['body'])>=100:
            over_100 = over_100 + 1
        if len(line_in['body'])>=150:
            over_150 = over_150 + 1
        if len(line_in['body'])>=200:
            over_200 = over_200 + 1
        if len(line_in['body'])>=250:
            over_250 = over_250 + 1
        if len(line_in['body'])>=300:
            over_300 = over_300 + 1   
                           
        num_posts = num_posts + 1
            
with open("../../Reddit_Data_trimmed/June_17/Android.txt", 'r') as file_in:
    for line in file_in:
        line_in = json.loads(line)
        num_chars = num_chars + len(line_in['body'])
        if len(line_in['body'])>=50:
            over_50 = over_50 + 1
        if len(line_in['body'])>=100:
            over_100 = over_100 + 1
        if len(line_in['body'])>=150:
            over_150 = over_150 + 1
        if len(line_in['body'])>=200:
            over_200 = over_200 + 1
        if len(line_in['body'])>=250:
            over_250 = over_250 + 1
        if len(line_in['body'])>=300:
            over_300 = over_300 + 1            

        num_posts = num_posts + 1

print(num_posts)
print("Average length of body: " + str(num_chars/num_posts))
print("number of posts over 50  characters: " + str(over_50))
print("number of posts over 100  characters: " + str(over_100))
print("number of posts over 150 characters: " + str(over_150))
print("number of posts over 200  characters: " + str(over_200))
print("number of posts over 250  characters: " + str(over_250))
print("number of posts over 300 characters: " + str(over_300))

434356
Average length of body: 182.29371529344593
number of posts over 50  characters: 319113
number of posts over 100  characters: 222110
number of posts over 150 characters: 156447
number of posts over 200  characters: 113629
number of posts over 250  characters: 85317
number of posts over 300 characters: 65906


In [12]:
num_posts = 0
num_chars = 0
over_50 = 0
over_100 = 0
over_150 = 0
over_200 = 0
over_250 = 0
over_300 = 0

with open("../../Reddit_Data_trimmed/September_17/boardgames.txt", 'r') as file_in:
    for line in file_in:
        line_in = json.loads(line)
        num_chars = num_chars + len(line_in['body'])
        if len(line_in['body'])>=50:
            over_50 = over_50 + 1
        if len(line_in['body'])>=100:
            over_100 = over_100 + 1
        if len(line_in['body'])>=150:
            over_150 = over_150 + 1
        if len(line_in['body'])>=200:
            over_200 = over_200 + 1
        if len(line_in['body'])>=250:
            over_250 = over_250 + 1
        if len(line_in['body'])>=300:
            over_300 = over_300 + 1            
        
        num_posts = num_posts + 1

with open("../../Reddit_Data_trimmed/August_17/boardgames.txt", 'r') as file_in:
    for line in file_in:
        line_in = json.loads(line)
        num_chars = num_chars + len(line_in['body'])
        if len(line_in['body'])>=50:
            over_50 = over_50 + 1
        if len(line_in['body'])>=100:
            over_100 = over_100 + 1
        if len(line_in['body'])>=150:
            over_150 = over_150 + 1
        if len(line_in['body'])>=200:
            over_200 = over_200 + 1
        if len(line_in['body'])>=250:
            over_250 = over_250 + 1
        if len(line_in['body'])>=300:
            over_300 = over_300 + 1   
                           
        num_posts = num_posts + 1
            
with open("../../Reddit_Data_trimmed/June_17/boardgames.txt", 'r') as file_in:
    for line in file_in:
        line_in = json.loads(line)
        num_chars = num_chars + len(line_in['body'])
        if len(line_in['body'])>=50:
            over_50 = over_50 + 1
        if len(line_in['body'])>=100:
            over_100 = over_100 + 1
        if len(line_in['body'])>=150:
            over_150 = over_150 + 1
        if len(line_in['body'])>=200:
            over_200 = over_200 + 1
        if len(line_in['body'])>=250:
            over_250 = over_250 + 1
        if len(line_in['body'])>=300:
            over_300 = over_300 + 1            

        num_posts = num_posts + 1

print(num_posts)
print("Average length of body: " + str(num_chars/num_posts))
print("number of posts over 50  characters: " + str(over_50))
print("number of posts over 100  characters: " + str(over_100))
print("number of posts over 150 characters: " + str(over_150))
print("number of posts over 200  characters: " + str(over_200))
print("number of posts over 250  characters: " + str(over_250))
print("number of posts over 300 characters: " + str(over_300))

184762
Average length of body: 254.62208138037042
number of posts over 50  characters: 148076
number of posts over 100  characters: 116951
number of posts over 150 characters: 91284
number of posts over 200  characters: 72144
number of posts over 250  characters: 57927
number of posts over 300 characters: 47144


In [13]:
num_posts = 0
num_chars = 0
over_50 = 0
over_100 = 0
over_150 = 0
over_200 = 0
over_250 = 0
over_300 = 0

with open("../../Reddit_Data_trimmed/September_17/hockey.txt", 'r') as file_in:
    for line in file_in:
        line_in = json.loads(line)
        num_chars = num_chars + len(line_in['body'])
        if len(line_in['body'])>=50:
            over_50 = over_50 + 1
        if len(line_in['body'])>=100:
            over_100 = over_100 + 1
        if len(line_in['body'])>=150:
            over_150 = over_150 + 1
        if len(line_in['body'])>=200:
            over_200 = over_200 + 1
        if len(line_in['body'])>=250:
            over_250 = over_250 + 1
        if len(line_in['body'])>=300:
            over_300 = over_300 + 1            
        
        num_posts = num_posts + 1

with open("../../Reddit_Data_trimmed/August_17/hockey.txt", 'r') as file_in:
    for line in file_in:
        line_in = json.loads(line)
        num_chars = num_chars + len(line_in['body'])
        if len(line_in['body'])>=50:
            over_50 = over_50 + 1
        if len(line_in['body'])>=100:
            over_100 = over_100 + 1
        if len(line_in['body'])>=150:
            over_150 = over_150 + 1
        if len(line_in['body'])>=200:
            over_200 = over_200 + 1
        if len(line_in['body'])>=250:
            over_250 = over_250 + 1
        if len(line_in['body'])>=300:
            over_300 = over_300 + 1   
                           
        num_posts = num_posts + 1
            
with open("../../Reddit_Data_trimmed/June_17/hockey.txt", 'r') as file_in:
    for line in file_in:
        line_in = json.loads(line)
        num_chars = num_chars + len(line_in['body'])
        if len(line_in['body'])>=50:
            over_50 = over_50 + 1
        if len(line_in['body'])>=100:
            over_100 = over_100 + 1
        if len(line_in['body'])>=150:
            over_150 = over_150 + 1
        if len(line_in['body'])>=200:
            over_200 = over_200 + 1
        if len(line_in['body'])>=250:
            over_250 = over_250 + 1
        if len(line_in['body'])>=300:
            over_300 = over_300 + 1            

        num_posts = num_posts + 1

print(num_posts)
print("Average length of body: " + str(num_chars/num_posts))
print("number of posts over 50  characters: " + str(over_50))
print("number of posts over 100  characters: " + str(over_100))
print("number of posts over 150 characters: " + str(over_150))
print("number of posts over 200  characters: " + str(over_200))
print("number of posts over 250  characters: " + str(over_250))
print("number of posts over 300 characters: " + str(over_300))

843259
Average length of body: 114.62820082560637
number of posts over 50  characters: 501472
number of posts over 100  characters: 285673
number of posts over 150 characters: 179058
number of posts over 200  characters: 120722
number of posts over 250  characters: 86133
number of posts over 300 characters: 63933


If I want to have atleast 50,000 posts from each subreddit, I will have to work with posts with lengths greater than 150 characters.  The subreddit with the least number of posts greater than 150 characters is "Conservative" with 69,784 posts.

In [15]:
d = {'parent_id':[], 
     'author':[], 
     'distinguished':[], 
     'body':[], 
     'gilded':[], 
     'score':[], 
     'author_flair_css_class':[],
     'stickied':[],
     'retrieved_on':[],
     'author_flair_text':[],
     'id':[]
    }

x = 0

temp_df = pd.DataFrame(data=d)

t = '../../csv_files'

with open("../../Reddit_Data_trimmed/September_17/Conservative.txt", 'r') as file_in:
    for line in file_in:
        line_in = json.loads(line)
        if x < 50000:
            if len(line_in['body']) >= 150:
                temp_df.loc[x, 'parent_id'] = line_in['parent_id']
                temp_df.loc[x, 'author'] = line_in['author']
                temp_df.loc[x, 'distinguished'] = line_in['distinguished']
                temp_df.loc[x, 'body'] = line_in['body']
                temp_df.loc[x, 'gilded'] = line_in['gilded']
                temp_df.loc[x, 'score'] = line_in['score']
                temp_df.loc[x, 'author_flair_css_class'] = line_in['author_flair_css_class']
                temp_df.loc[x, 'stickied'] = line_in['stickied']
                temp_df.loc[x, 'retrieved_on'] = line_in['retrieved_on']
                temp_df.loc[x, 'author_flair_text'] = line_in['author_flair_text']
                temp_df.loc[x, 'id'] = line_in['id']
                x = x +1
        else:
            break
            
with open("../../Reddit_Data_trimmed/August_17/Conservative.txt", 'r') as file_in:
    for line in file_in:
        line_in = json.loads(line)
        if x < 50000:
            if len(line_in['body']) >= 150:
                temp_df.loc[x, 'parent_id'] = line_in['parent_id']
                temp_df.loc[x, 'author'] = line_in['author']
                temp_df.loc[x, 'distinguished'] = line_in['distinguished']
                temp_df.loc[x, 'body'] = line_in['body']
                temp_df.loc[x, 'gilded'] = line_in['gilded']
                temp_df.loc[x, 'score'] = line_in['score']
                temp_df.loc[x, 'author_flair_css_class'] = line_in['author_flair_css_class']
                temp_df.loc[x, 'stickied'] = line_in['stickied']
                temp_df.loc[x, 'retrieved_on'] = line_in['retrieved_on']
                temp_df.loc[x, 'author_flair_text'] = line_in['author_flair_text']
                temp_df.loc[x, 'id'] = line_in['id']
                x = x +1
        else:
            break
        
with open("../../Reddit_Data_trimmed/June_17/Conservative.txt", 'r') as file_in:
    for line in file_in:
        line_in = json.loads(line)
        if x < 50000:
            if len(line_in['body']) >= 150:
                temp_df.loc[x, 'parent_id'] = line_in['parent_id']
                temp_df.loc[x, 'author'] = line_in['author']
                temp_df.loc[x, 'distinguished'] = line_in['distinguished']
                temp_df.loc[x, 'body'] = line_in['body']
                temp_df.loc[x, 'gilded'] = line_in['gilded']
                temp_df.loc[x, 'score'] = line_in['score']
                temp_df.loc[x, 'author_flair_css_class'] = line_in['author_flair_css_class']
                temp_df.loc[x, 'stickied'] = line_in['stickied']
                temp_df.loc[x, 'retrieved_on'] = line_in['retrieved_on']
                temp_df.loc[x, 'author_flair_text'] = line_in['author_flair_text']
                temp_df.loc[x, 'id'] = line_in['id']
                x = x +1
        else:
            break
    
temp_df.to_csv(t + '/Conservative.csv', sep='\t')

In [16]:
x = 0

with open("../../Reddit_Data_trimmed/September_17/Libertarian.txt", 'r') as file_in:
    for line in file_in:
        line_in = json.loads(line)
        if x < 50000:
            if len(line_in['body']) >= 150:
                temp_df.loc[x, 'parent_id'] = line_in['parent_id']
                temp_df.loc[x, 'author'] = line_in['author']
                temp_df.loc[x, 'distinguished'] = line_in['distinguished']
                temp_df.loc[x, 'body'] = line_in['body']
                temp_df.loc[x, 'gilded'] = line_in['gilded']
                temp_df.loc[x, 'score'] = line_in['score']
                temp_df.loc[x, 'author_flair_css_class'] = line_in['author_flair_css_class']
                temp_df.loc[x, 'stickied'] = line_in['stickied']
                temp_df.loc[x, 'retrieved_on'] = line_in['retrieved_on']
                temp_df.loc[x, 'author_flair_text'] = line_in['author_flair_text']
                temp_df.loc[x, 'id'] = line_in['id']
                x = x +1
        else:
            break
            
with open("../../Reddit_Data_trimmed/August_17/Libertarian.txt", 'r') as file_in:
    for line in file_in:
        line_in = json.loads(line)
        if x < 50000:
            if len(line_in['body']) >= 150:
                temp_df.loc[x, 'parent_id'] = line_in['parent_id']
                temp_df.loc[x, 'author'] = line_in['author']
                temp_df.loc[x, 'distinguished'] = line_in['distinguished']
                temp_df.loc[x, 'body'] = line_in['body']
                temp_df.loc[x, 'gilded'] = line_in['gilded']
                temp_df.loc[x, 'score'] = line_in['score']
                temp_df.loc[x, 'author_flair_css_class'] = line_in['author_flair_css_class']
                temp_df.loc[x, 'stickied'] = line_in['stickied']
                temp_df.loc[x, 'retrieved_on'] = line_in['retrieved_on']
                temp_df.loc[x, 'author_flair_text'] = line_in['author_flair_text']
                temp_df.loc[x, 'id'] = line_in['id']
                x = x +1
        else:
            break
        
with open("../../Reddit_Data_trimmed/June_17/Libertarian.txt", 'r') as file_in:
    for line in file_in:
        line_in = json.loads(line)
        if x < 50000:
            if len(line_in['body']) >= 150:
                temp_df.loc[x, 'parent_id'] = line_in['parent_id']
                temp_df.loc[x, 'author'] = line_in['author']
                temp_df.loc[x, 'distinguished'] = line_in['distinguished']
                temp_df.loc[x, 'body'] = line_in['body']
                temp_df.loc[x, 'gilded'] = line_in['gilded']
                temp_df.loc[x, 'score'] = line_in['score']
                temp_df.loc[x, 'author_flair_css_class'] = line_in['author_flair_css_class']
                temp_df.loc[x, 'stickied'] = line_in['stickied']
                temp_df.loc[x, 'retrieved_on'] = line_in['retrieved_on']
                temp_df.loc[x, 'author_flair_text'] = line_in['author_flair_text']
                temp_df.loc[x, 'id'] = line_in['id']
        else:
            break

temp_df.to_csv(t + '/Libertarian.csv', sep='\t')

In [17]:
x = 0

with open("../../Reddit_Data_trimmed/September_17/neoliberal.txt", 'r') as file_in:
    for line in file_in:
        line_in = json.loads(line)
        if x < 50000:
            if len(line_in['body']) >= 150:
                temp_df.loc[x, 'parent_id'] = line_in['parent_id']
                temp_df.loc[x, 'author'] = line_in['author']
                temp_df.loc[x, 'distinguished'] = line_in['distinguished']
                temp_df.loc[x, 'body'] = line_in['body']
                temp_df.loc[x, 'gilded'] = line_in['gilded']
                temp_df.loc[x, 'score'] = line_in['score']
                temp_df.loc[x, 'author_flair_css_class'] = line_in['author_flair_css_class']
                temp_df.loc[x, 'stickied'] = line_in['stickied']
                temp_df.loc[x, 'retrieved_on'] = line_in['retrieved_on']
                temp_df.loc[x, 'author_flair_text'] = line_in['author_flair_text']
                temp_df.loc[x, 'id'] = line_in['id']
                x = x +1
        else:
            break
            
with open("../../Reddit_Data_trimmed/August_17/neoliberal.txt", 'r') as file_in:
    for line in file_in:
        line_in = json.loads(line)
        if x < 50000:
            if len(line_in['body']) >= 150:
                temp_df.loc[x, 'parent_id'] = line_in['parent_id']
                temp_df.loc[x, 'author'] = line_in['author']
                temp_df.loc[x, 'distinguished'] = line_in['distinguished']
                temp_df.loc[x, 'body'] = line_in['body']
                temp_df.loc[x, 'gilded'] = line_in['gilded']
                temp_df.loc[x, 'score'] = line_in['score']
                temp_df.loc[x, 'author_flair_css_class'] = line_in['author_flair_css_class']
                temp_df.loc[x, 'stickied'] = line_in['stickied']
                temp_df.loc[x, 'retrieved_on'] = line_in['retrieved_on']
                temp_df.loc[x, 'author_flair_text'] = line_in['author_flair_text']
                temp_df.loc[x, 'id'] = line_in['id']
                x = x +1
        else:
            break
        
with open("../../Reddit_Data_trimmed/June_17/neoliberal.txt", 'r') as file_in:
    for line in file_in:
        line_in = json.loads(line)
        if x < 50000:
            if len(line_in['body']) >= 150:
                temp_df.loc[x, 'parent_id'] = line_in['parent_id']
                temp_df.loc[x, 'author'] = line_in['author']
                temp_df.loc[x, 'distinguished'] = line_in['distinguished']
                temp_df.loc[x, 'body'] = line_in['body']
                temp_df.loc[x, 'gilded'] = line_in['gilded']
                temp_df.loc[x, 'score'] = line_in['score']
                temp_df.loc[x, 'author_flair_css_class'] = line_in['author_flair_css_class']
                temp_df.loc[x, 'stickied'] = line_in['stickied']
                temp_df.loc[x, 'retrieved_on'] = line_in['retrieved_on']
                temp_df.loc[x, 'author_flair_text'] = line_in['author_flair_text']
                temp_df.loc[x, 'id'] = line_in['id']
        else:
            break
    
temp_df.to_csv(t + '/neoliberal.csv', sep='\t')

In [18]:
x = 0

with open("../../Reddit_Data_trimmed/September_17/politics.txt", 'r') as file_in:
    for line in file_in:
        line_in = json.loads(line)
        if x < 50000:
            if len(line_in['body']) >= 150:
                temp_df.loc[x, 'parent_id'] = line_in['parent_id']
                temp_df.loc[x, 'author'] = line_in['author']
                temp_df.loc[x, 'distinguished'] = line_in['distinguished']
                temp_df.loc[x, 'body'] = line_in['body']
                temp_df.loc[x, 'gilded'] = line_in['gilded']
                temp_df.loc[x, 'score'] = line_in['score']
                temp_df.loc[x, 'author_flair_css_class'] = line_in['author_flair_css_class']
                temp_df.loc[x, 'stickied'] = line_in['stickied']
                temp_df.loc[x, 'retrieved_on'] = line_in['retrieved_on']
                temp_df.loc[x, 'author_flair_text'] = line_in['author_flair_text']
                temp_df.loc[x, 'id'] = line_in['id']
                x = x +1
        else:
            break
            
with open("../../Reddit_Data_trimmed/August_17/politics.txt", 'r') as file_in:
    for line in file_in:
        line_in = json.loads(line)
        if x < 50000:
            if len(line_in['body']) >= 150:
                temp_df.loc[x, 'parent_id'] = line_in['parent_id']
                temp_df.loc[x, 'author'] = line_in['author']
                temp_df.loc[x, 'distinguished'] = line_in['distinguished']
                temp_df.loc[x, 'body'] = line_in['body']
                temp_df.loc[x, 'gilded'] = line_in['gilded']
                temp_df.loc[x, 'score'] = line_in['score']
                temp_df.loc[x, 'author_flair_css_class'] = line_in['author_flair_css_class']
                temp_df.loc[x, 'stickied'] = line_in['stickied']
                temp_df.loc[x, 'retrieved_on'] = line_in['retrieved_on']
                temp_df.loc[x, 'author_flair_text'] = line_in['author_flair_text']
                temp_df.loc[x, 'id'] = line_in['id']
                x = x +1
        else:
            break
        
with open("../../Reddit_Data_trimmed/June_17/politics.txt", 'r') as file_in:
    for line in file_in:
        line_in = json.loads(line)
        if x < 50000:
            if len(line_in['body']) >= 150:
                temp_df.loc[x, 'parent_id'] = line_in['parent_id']
                temp_df.loc[x, 'author'] = line_in['author']
                temp_df.loc[x, 'distinguished'] = line_in['distinguished']
                temp_df.loc[x, 'body'] = line_in['body']
                temp_df.loc[x, 'gilded'] = line_in['gilded']
                temp_df.loc[x, 'score'] = line_in['score']
                temp_df.loc[x, 'author_flair_css_class'] = line_in['author_flair_css_class']
                temp_df.loc[x, 'stickied'] = line_in['stickied']
                temp_df.loc[x, 'retrieved_on'] = line_in['retrieved_on']
                temp_df.loc[x, 'author_flair_text'] = line_in['author_flair_text']
                temp_df.loc[x, 'id'] = line_in['id']
        else:
            break

temp_df.to_csv(t + '/politics.csv', sep='\t')

In [19]:
x = 0

with open("../../Reddit_Data_trimmed/September_17/worldnews.txt", 'r') as file_in:
    for line in file_in:
        line_in = json.loads(line)
        if x < 50000:
            if len(line_in['body']) >= 150:
                temp_df.loc[x, 'parent_id'] = line_in['parent_id']
                temp_df.loc[x, 'author'] = line_in['author']
                temp_df.loc[x, 'distinguished'] = line_in['distinguished']
                temp_df.loc[x, 'body'] = line_in['body']
                temp_df.loc[x, 'gilded'] = line_in['gilded']
                temp_df.loc[x, 'score'] = line_in['score']
                temp_df.loc[x, 'author_flair_css_class'] = line_in['author_flair_css_class']
                temp_df.loc[x, 'stickied'] = line_in['stickied']
                temp_df.loc[x, 'retrieved_on'] = line_in['retrieved_on']
                temp_df.loc[x, 'author_flair_text'] = line_in['author_flair_text']
                temp_df.loc[x, 'id'] = line_in['id']
                x = x +1
        else:
            break
            
with open("../../Reddit_Data_trimmed/August_17/worldnews.txt", 'r') as file_in:
    for line in file_in:
        line_in = json.loads(line)
        if x < 50000:
            if len(line_in['body']) >= 150:
                temp_df.loc[x, 'parent_id'] = line_in['parent_id']
                temp_df.loc[x, 'author'] = line_in['author']
                temp_df.loc[x, 'distinguished'] = line_in['distinguished']
                temp_df.loc[x, 'body'] = line_in['body']
                temp_df.loc[x, 'gilded'] = line_in['gilded']
                temp_df.loc[x, 'score'] = line_in['score']
                temp_df.loc[x, 'author_flair_css_class'] = line_in['author_flair_css_class']
                temp_df.loc[x, 'stickied'] = line_in['stickied']
                temp_df.loc[x, 'retrieved_on'] = line_in['retrieved_on']
                temp_df.loc[x, 'author_flair_text'] = line_in['author_flair_text']
                temp_df.loc[x, 'id'] = line_in['id']
                x = x +1
        else:
            break
        
with open("../../Reddit_Data_trimmed/June_17/worldnews.txt", 'r') as file_in:
    for line in file_in:
        line_in = json.loads(line)
        if x < 50000:
            if len(line_in['body']) >= 150:
                temp_df.loc[x, 'parent_id'] = line_in['parent_id']
                temp_df.loc[x, 'author'] = line_in['author']
                temp_df.loc[x, 'distinguished'] = line_in['distinguished']
                temp_df.loc[x, 'body'] = line_in['body']
                temp_df.loc[x, 'gilded'] = line_in['gilded']
                temp_df.loc[x, 'score'] = line_in['score']
                temp_df.loc[x, 'author_flair_css_class'] = line_in['author_flair_css_class']
                temp_df.loc[x, 'stickied'] = line_in['stickied']
                temp_df.loc[x, 'retrieved_on'] = line_in['retrieved_on']
                temp_df.loc[x, 'author_flair_text'] = line_in['author_flair_text']
                temp_df.loc[x, 'id'] = line_in['id']
        else:
            break

temp_df.to_csv(t + '/worldnews.csv', sep='\t')

In [20]:
x = 0

with open("../../Reddit_Data_trimmed/September_17/Android.txt", 'r') as file_in:
    for line in file_in:
        line_in = json.loads(line)
        if x < 50000:
            if len(line_in['body']) >= 150:
                temp_df.loc[x, 'parent_id'] = line_in['parent_id']
                temp_df.loc[x, 'author'] = line_in['author']
                temp_df.loc[x, 'distinguished'] = line_in['distinguished']
                temp_df.loc[x, 'body'] = line_in['body']
                temp_df.loc[x, 'gilded'] = line_in['gilded']
                temp_df.loc[x, 'score'] = line_in['score']
                temp_df.loc[x, 'author_flair_css_class'] = line_in['author_flair_css_class']
                temp_df.loc[x, 'stickied'] = line_in['stickied']
                temp_df.loc[x, 'retrieved_on'] = line_in['retrieved_on']
                temp_df.loc[x, 'author_flair_text'] = line_in['author_flair_text']
                temp_df.loc[x, 'id'] = line_in['id']
                x = x +1
        else:
            break
            
with open("../../Reddit_Data_trimmed/August_17/Android.txt", 'r') as file_in:
    for line in file_in:
        line_in = json.loads(line)
        if x < 50000:
            if len(line_in['body']) >= 150:
                temp_df.loc[x, 'parent_id'] = line_in['parent_id']
                temp_df.loc[x, 'author'] = line_in['author']
                temp_df.loc[x, 'distinguished'] = line_in['distinguished']
                temp_df.loc[x, 'body'] = line_in['body']
                temp_df.loc[x, 'gilded'] = line_in['gilded']
                temp_df.loc[x, 'score'] = line_in['score']
                temp_df.loc[x, 'author_flair_css_class'] = line_in['author_flair_css_class']
                temp_df.loc[x, 'stickied'] = line_in['stickied']
                temp_df.loc[x, 'retrieved_on'] = line_in['retrieved_on']
                temp_df.loc[x, 'author_flair_text'] = line_in['author_flair_text']
                temp_df.loc[x, 'id'] = line_in['id']
                x = x +1
        else:
            break
        
with open("../../Reddit_Data_trimmed/June_17/Android.txt", 'r') as file_in:
    for line in file_in:
        line_in = json.loads(line)
        if x < 50000:
            if len(line_in['body']) >= 150:
                temp_df.loc[x, 'parent_id'] = line_in['parent_id']
                temp_df.loc[x, 'author'] = line_in['author']
                temp_df.loc[x, 'distinguished'] = line_in['distinguished']
                temp_df.loc[x, 'body'] = line_in['body']
                temp_df.loc[x, 'gilded'] = line_in['gilded']
                temp_df.loc[x, 'score'] = line_in['score']
                temp_df.loc[x, 'author_flair_css_class'] = line_in['author_flair_css_class']
                temp_df.loc[x, 'stickied'] = line_in['stickied']
                temp_df.loc[x, 'retrieved_on'] = line_in['retrieved_on']
                temp_df.loc[x, 'author_flair_text'] = line_in['author_flair_text']
                temp_df.loc[x, 'id'] = line_in['id']
        else:
            break

temp_df.to_csv(t + '/Android.csv', sep='\t')

In [21]:
x = 0

with open("../../Reddit_Data_trimmed/September_17/boardgames.txt", 'r') as file_in:
    for line in file_in:
        line_in = json.loads(line)
        if x < 50000:
            if len(line_in['body']) >= 150:
                temp_df.loc[x, 'parent_id'] = line_in['parent_id']
                temp_df.loc[x, 'author'] = line_in['author']
                temp_df.loc[x, 'distinguished'] = line_in['distinguished']
                temp_df.loc[x, 'body'] = line_in['body']
                temp_df.loc[x, 'gilded'] = line_in['gilded']
                temp_df.loc[x, 'score'] = line_in['score']
                temp_df.loc[x, 'author_flair_css_class'] = line_in['author_flair_css_class']
                temp_df.loc[x, 'stickied'] = line_in['stickied']
                temp_df.loc[x, 'retrieved_on'] = line_in['retrieved_on']
                temp_df.loc[x, 'author_flair_text'] = line_in['author_flair_text']
                temp_df.loc[x, 'id'] = line_in['id']
                x = x +1
        else:
            break
            
with open("../../Reddit_Data_trimmed/August_17/boardgames.txt", 'r') as file_in:
    for line in file_in:
        line_in = json.loads(line)
        if x < 50000:
            if len(line_in['body']) >= 150:
                temp_df.loc[x, 'parent_id'] = line_in['parent_id']
                temp_df.loc[x, 'author'] = line_in['author']
                temp_df.loc[x, 'distinguished'] = line_in['distinguished']
                temp_df.loc[x, 'body'] = line_in['body']
                temp_df.loc[x, 'gilded'] = line_in['gilded']
                temp_df.loc[x, 'score'] = line_in['score']
                temp_df.loc[x, 'author_flair_css_class'] = line_in['author_flair_css_class']
                temp_df.loc[x, 'stickied'] = line_in['stickied']
                temp_df.loc[x, 'retrieved_on'] = line_in['retrieved_on']
                temp_df.loc[x, 'author_flair_text'] = line_in['author_flair_text']
                temp_df.loc[x, 'id'] = line_in['id']
                x = x +1
        else:
            break
        
with open("../../Reddit_Data_trimmed/June_17/boardgames.txt", 'r') as file_in:
    for line in file_in:
        line_in = json.loads(line)
        if x < 50000:
            if len(line_in['body']) >= 150:
                temp_df.loc[x, 'parent_id'] = line_in['parent_id']
                temp_df.loc[x, 'author'] = line_in['author']
                temp_df.loc[x, 'distinguished'] = line_in['distinguished']
                temp_df.loc[x, 'body'] = line_in['body']
                temp_df.loc[x, 'gilded'] = line_in['gilded']
                temp_df.loc[x, 'score'] = line_in['score']
                temp_df.loc[x, 'author_flair_css_class'] = line_in['author_flair_css_class']
                temp_df.loc[x, 'stickied'] = line_in['stickied']
                temp_df.loc[x, 'retrieved_on'] = line_in['retrieved_on']
                temp_df.loc[x, 'author_flair_text'] = line_in['author_flair_text']
                temp_df.loc[x, 'id'] = line_in['id']
        else:
            break

temp_df.to_csv(t + '/boardgames.csv', sep='\t')

In [22]:
x = 0

with open("../../Reddit_Data_trimmed/September_17/hockey.txt", 'r') as file_in:
    for line in file_in:
        line_in = json.loads(line)
        if x < 50000:
            if len(line_in['body']) >= 150:
                temp_df.loc[x, 'parent_id'] = line_in['parent_id']
                temp_df.loc[x, 'author'] = line_in['author']
                temp_df.loc[x, 'distinguished'] = line_in['distinguished']
                temp_df.loc[x, 'body'] = line_in['body']
                temp_df.loc[x, 'gilded'] = line_in['gilded']
                temp_df.loc[x, 'score'] = line_in['score']
                temp_df.loc[x, 'author_flair_css_class'] = line_in['author_flair_css_class']
                temp_df.loc[x, 'stickied'] = line_in['stickied']
                temp_df.loc[x, 'retrieved_on'] = line_in['retrieved_on']
                temp_df.loc[x, 'author_flair_text'] = line_in['author_flair_text']
                temp_df.loc[x, 'id'] = line_in['id']
                x = x +1
        else:
            break
            
with open("../../Reddit_Data_trimmed/August_17/hockey.txt", 'r') as file_in:
    for line in file_in:
        line_in = json.loads(line)
        if x < 50000:
            if len(line_in['body']) >= 150:
                temp_df.loc[x, 'parent_id'] = line_in['parent_id']
                temp_df.loc[x, 'author'] = line_in['author']
                temp_df.loc[x, 'distinguished'] = line_in['distinguished']
                temp_df.loc[x, 'body'] = line_in['body']
                temp_df.loc[x, 'gilded'] = line_in['gilded']
                temp_df.loc[x, 'score'] = line_in['score']
                temp_df.loc[x, 'author_flair_css_class'] = line_in['author_flair_css_class']
                temp_df.loc[x, 'stickied'] = line_in['stickied']
                temp_df.loc[x, 'retrieved_on'] = line_in['retrieved_on']
                temp_df.loc[x, 'author_flair_text'] = line_in['author_flair_text']
                temp_df.loc[x, 'id'] = line_in['id']
                x = x +1
        else:
            break
        
with open("../../Reddit_Data_trimmed/June_17/hockey.txt", 'r') as file_in:
    for line in file_in:
        line_in = json.loads(line)
        if x < 50000:
            if len(line_in['body']) >= 150:
                temp_df.loc[x, 'parent_id'] = line_in['parent_id']
                temp_df.loc[x, 'author'] = line_in['author']
                temp_df.loc[x, 'distinguished'] = line_in['distinguished']
                temp_df.loc[x, 'body'] = line_in['body']
                temp_df.loc[x, 'gilded'] = line_in['gilded']
                temp_df.loc[x, 'score'] = line_in['score']
                temp_df.loc[x, 'author_flair_css_class'] = line_in['author_flair_css_class']
                temp_df.loc[x, 'stickied'] = line_in['stickied']
                temp_df.loc[x, 'retrieved_on'] = line_in['retrieved_on']
                temp_df.loc[x, 'author_flair_text'] = line_in['author_flair_text']
                temp_df.loc[x, 'id'] = line_in['id']
        else:
            break

temp_df.to_csv(t + '/hockey.csv', sep='\t')